### PR TITLE
fix building on windows with vs2019

### PR DESCRIPTION
**Pull request by blagoev**
Fixes building on windows with vs2019

### CODE REVIEW

File: src/util/event_loop_dispatcher.hpp [hunk 1]
```diff
@@ -54,11 +54,11 @@ class EventLoopDispatcher<void(Args...)> {
     EventLoopDispatcher(std::function<void(Args...)> func)
     : m_state(std::make_shared<State>(std::move(func)))
     {
-        m_scheduler->set_notify_callback([state = m_state] {
+        m_scheduler->set_notify_callback([state = m_state, this] {
```
Comment (fealebenpae) [inline src/util/event_loop_dispatcher.hpp:57]:
```suggestion
        m_scheduler->set_notify_callback([state = m_state] {
```

File: src/util/event_loop_dispatcher.hpp [hunk 2]
```diff
@@ -54,11 +54,11 @@ class EventLoopDispatcher<void(Args...)> {
...
             std::unique_lock<std::mutex> lock(state->mutex);
             while (!state->invocations.empty()) {
                 auto& tuple = state->invocations.front();
-                _apply_polyfill::apply(std::move(tuple), state->func);
+                std::apply(m_state->func, std::move(tuple));
```
Comment (fealebenpae) [inline src/util/event_loop_dispatcher.hpp:61]:
```suggestion
                std::apply(state->func, std::move(tuple));
```

File: src/util/event_loop_dispatcher.hpp (hunk 1 above)
Comment (blagoev) [reply src/util/event_loop_dispatcher.hpp]: good catch. It seems this file was not merged correctly.

File: src/sync/app.cpp [hunk 3]
```diff
@@ -910,13 +909,13 @@ void App::call_function(std::shared_ptr<SyncUser> user,
...
 
-    Request request {
-        .method = HttpMethod::post,
-        .url = route,
-        .body = s.str()
-    };
+    Request req;
+    req.method = HttpMethod::post;
+    req.url = route;
+    req.body = s.str();
```
Comment (ironage) [inline src/sync/app.cpp:915]: It looks like this change caused build failures in the app tests on the v10 branch. The previous way that the Request was initialized would zero initialize the omitted members, namely `uses_refresh_token` should be false here. The linux app tests did fail on this PR in CI. I can see that this error was obscured by a separate windows issue, but please be careful when merging PR's which fail CI.

Comment (blagoev) [reply src/sync/app.cpp:915]: sorry about that

**Pull request merged**